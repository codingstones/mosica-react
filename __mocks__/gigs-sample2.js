export const sampleGigs = [
  {
    'day': 'Sábado, 20 de Agosto',
    'gigs': [
      {
        'title': 'rapsus',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'cesareo alierta, 5001,',
        'description': 'Concierto de rapsus',
        'price': '',
        'image_url': 'http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png',
        'affiliate_url': null,
        'place': 'Avenida Cesáreo Alierta (mobilityweek)',
        'slug': 'javat',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-09-18',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      }
    ]
  }
]
