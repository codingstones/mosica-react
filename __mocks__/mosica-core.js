import { sampleGigs } from './gigs-sample'
const mosicaCore = jest.genMockFromModule('mosica-core')

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(sampleGigs)
  }
}

mosicaCore.GigService = GigService
module.exports = mosicaCore
