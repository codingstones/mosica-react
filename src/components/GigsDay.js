import React from 'react';
import { Gigs } from './Gigs';

export const GigsDay = (props) => {
  const {days, onClick} = props;
  return (
    <div>
      {days.map((day) =>
        <div className="gigs" key={day.day}>
          <div className="day">
            {day.day}
          </div>
          <Gigs gigs={day.gigs} onClick={onClick}/>
        </div>
      )}
    </div>
  );
};