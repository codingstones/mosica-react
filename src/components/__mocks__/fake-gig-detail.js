export const fakeGigDetail = {
  address: "Avda. Cesáreo Alierta, 120, 50013, Zaragoza",
  affiliate_url: null,
  day: "2017-12-16",
  description:
    "Enrique Bunbury ha anunciado el lanzamiento de su nuevo disco de estudio Expectativas. Hará una parada en Zaragoza el próximo 16 de diciembre para presentarlo ante sus paisanos. Este nuevo álbum es la continuación de Palosanto de 2013. La espera ha sido de cuatro años, pero el 8 de septiembre llegarán por fin los dos primeros adelantos,",
  hour: "22:00",
  id: 6533,
  image_url: "http://www.zaragoza.es/cont/paginas/actividades/imagen/pfotoevento4020.png",
  lat_lng: "41.63602275205834,-0.8658168875758695",
  place: "Pabellón Príncipe Felipe",
  price: "",
  schedule: [],
  schedule_id: 9119,
  slug: "concierto-enrique-bunbury",
  title: "Concierto Enrique Bunbury"
}