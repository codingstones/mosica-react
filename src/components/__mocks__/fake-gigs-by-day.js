export const fakeGigsByDay = [
  {
    'day': 'Lunes, 18 de Septiembre',
    'gigs': [
      {
        'id': '123456',
        'title': 'Anarchy in the JS',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png',
        'affiliate_url': null,
        'place': 'SCBCN',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-09-18',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
      {
        'id': '2222222',
        'title': 'Anarchy in the Codemotion',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Another Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.wtf.es/wtf.png',
        'affiliate_url': null,
        'place': 'Around the world',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-12-03',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
    ]
  }
]

export const FIRST_DAY_GIG_TEXTS = fakeGigsByDay[0].gigs.map(gig => gig.title)
export const FIRST_GIG = fakeGigsByDay[0].gigs[0]
