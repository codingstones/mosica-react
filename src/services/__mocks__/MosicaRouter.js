const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MosicaRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MosicaRouter
}

