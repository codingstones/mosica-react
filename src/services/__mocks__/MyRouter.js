const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MyRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MyRouter: MyRouter,
}

