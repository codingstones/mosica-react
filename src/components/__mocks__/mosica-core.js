import { fakeGigsByDay } from './fake-gigs-by-day'
const mosicaCore = jest.genMockFromModule('mosica-core')

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(fakeGigsByDay)
  }
}

mosicaCore.GigService = GigService
module.exports = mosicaCore
