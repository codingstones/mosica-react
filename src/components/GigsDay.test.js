import React from 'react';
import renderer from 'react-test-renderer';
import { GigsDay } from './GigsDay';
import {fakeGigsByDay} from './__mocks__/fake-gigs-by-day';

it('renders GigsDay', () => {
  const component = renderer.create(<GigsDay days={fakeGigsByDay}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

