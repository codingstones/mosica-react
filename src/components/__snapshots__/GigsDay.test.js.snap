// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GigsDay 1`] = `
<div>
  <div
    className="gigs"
  >
    <a
      className="day"
      href={
        Object {
          "day": "Lunes, 18 de Septiembre",
          "gigs": Array [
            Object {
              "address": "BCN",
              "affiliate_url": null,
              "day": "2017-09-18",
              "description": "Coding Stones Gig",
              "hour": "19:30",
              "id": "123456",
              "image_url": "http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png",
              "lat_lng": "41.641851935961654,-0.8751129897638315",
              "place": "SCBCN",
              "price": "",
              "schedule": Array [
                Object {
                  "created_at": "2017-09-07T15:57:35.419Z",
                  "day": "2017-09-18",
                  "gig_id": 5927,
                  "hour": "19:30",
                  "id": 5280,
                  "updated_at": "2017-09-07T15:57:35.419Z",
                },
              ],
              "schedule_id": 5280,
              "slug": "stones",
              "title": "Anarchy in the JS",
            },
            Object {
              "address": "BCN",
              "affiliate_url": null,
              "day": "2017-12-03",
              "description": "Another Coding Stones Gig",
              "hour": "19:30",
              "id": "2222222",
              "image_url": "http://www.wtf.es/wtf.png",
              "lat_lng": "41.641851935961654,-0.8751129897638315",
              "place": "Around the world",
              "price": "",
              "schedule": Array [
                Object {
                  "created_at": "2017-09-07T15:57:35.419Z",
                  "day": "2017-09-18",
                  "gig_id": 5927,
                  "hour": "19:30",
                  "id": 5280,
                  "updated_at": "2017-09-07T15:57:35.419Z",
                },
              ],
              "schedule_id": 5280,
              "slug": "stones",
              "title": "Vegan Microservices",
            },
          ],
        }
      }
      target="_blank"
    >
      Lunes, 18 de Septiembre
    </a>
    <div>
      <div
        className="row"
      >
        Anarchy in the JS
        -
        SCBCN
      </div>
      <div
        className="row"
      >
        Vegan Microservices
        -
        Around the world
      </div>
    </div>
  </div>
</div>
`;
